### ISSUE TITLE
[RFC] Is the app supposed to be run in the foreground

### ISSUE THREAD

**Issue by rickysarraf** [ankitstarski/GeoclueShare#5]
If I need the GPS co-ordinates in my laptop, do I need to start the app on my phone ?

RIght now, when started, I have a static page with "Location Mode", "Location Sharing" and Connected Devices. Does this app need to be in the forground ? Or can it be backgrounded ?

On that note, why run the application? Can't the app register a listener, and then when it gets a request, it could wake up, get the data and feed it to the requestor.

Sorry it these questions have been asked before. I tried to look for documentation but could not find much.

**Comment by ankitstarski** [comment]
No, the app does not need to be in the foreground for location sharing (you might have to turn on location sharing though).

The reason that we are not doing that listener thing is because we are using Android application as a server and if the server is turned on all the time, it will consume battery.

If your questions are answered, please tell me so that I may close this issue. :)

**Comment by rickysarraf** [comment]
On Sun, 2015-09-27 at 08:34 -0700, Ankit wrote:

> No, the app does not need to be in the foreground for location
> sharing (you might have to turn on location sharing though).
> The reason that we are not doing that listener thing is because we
> are using Android application as a server and if the server is turned
> on all the time, it will consume battery.
> If your questions are answered, please tell me so that I may close
> this issue. :)

Yes. That was exactly my point. Not to have the server running all the
time. So, thus, you could have a listener registered with Android, and
when you get a request, you could spawn the server in a separate
thread, fetch the data, and provide it to the client. And then go back
to sleep, conserving the power/battery on the phone.

Or is Geoclueshare + GNOME GPS is supposed to work in broadcast mode ?
From what you mentioned in the other bug, that's what it sounds like.

## 

Ritesh Raj Sarraf
RESEARCHUT - http://www.researchut.com
"Necessity is the mother of invention."

**Comment by ankitstarski** [comment]
Yes, the location is broadcasted to all the clients connected to GeoclueShare.

**Comment by zeenix** [comment]
@ankitstarski i think you can safely close this issue.
